### PR TITLE
Systemctl kexec buglet fixes

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -3511,9 +3511,11 @@ static int load_kexec_kernel(void) {
         if (!options)
                 return log_oom();
 
-        log_debug("%s kexec kernel %s initrd %s options \"%s\".",
-                  arg_dry_run ? "Would load" : "loading",
-                  kernel, initrd, options);
+        log_info("%s "KEXEC" --load \"%s\" --append \"%s\"%s%s%s",
```
Comment (poettering) [inline src/systemctl/systemctl.c]: shouldn't this be:

```c
log_full(arg_quiet ? LOG_DEBUG : LOG_INFO, …
```

File: src/systemctl/systemctl.c [hunk 2]
```diff
@@ -3536,7 +3535,13 @@ static int load_kexec_kernel(void) {
                 _exit(EXIT_FAILURE);
         }
 
-        return wait_for_terminate_and_check("kexec", pid, WAIT_LOG);
+        r = wait_for_terminate_and_check("kexec", pid, WAIT_LOG);
+        if (r < 0)
+                return r;
+        if (r > 0)
+                /* Command failed */
+                return -EINVAL;
```
Comment (poettering) [inline src/systemctl/systemctl.c]: not that it matters, but we usually use EPROTO for propagating failures due to non-zero child process return values...

File: src/systemctl/systemctl.c [hunk 3]
```diff
@@ -3602,7 +3602,9 @@ static int start_special(int argc, char *argv[], void *userdata) {
 
         } else if (a == ACTION_KEXEC) {
                 r = load_kexec_kernel();
-                if (r < 0)
+                if (r < 0 && arg_force >= 1)
+                        log_notice("Failed to load kexec kernel, continuing without.");
+                else if (r < 0)
```
Comment (poettering) [inline src/systemctl/systemctl.c:3616]: not that it matters, but you check `r < 0` twice here... maybe:

```c
if (r < 0) {
        if (arg_force == 0) /* propagate the failure unless -f is given */
            return r;

        log_full(arg_quiet ? LOG_DEBUG : LOG_NOTICE, "…
}
```

**Comment by poettering** [comment]
Looks good. Only found some minor nitpicks regarding logging…

File: src/systemctl/systemctl.c (hunk 3 above)
Comment (keszybz) [reply src/systemctl/systemctl.c:3616]: I know ;) I did it this way to avoid spurious indentation. I'm sure the compiler can figure this out.

**Comment by keszybz** [comment]
Updated with the two suggested logging changes.

**Pull request merged**